### ISSUE TITLE
Extract pdm-plugin-torch?

### ISSUE THREAD

**Issue by dsully** [EmbarkStudios/emote#70]
Hi - I have need of `pdm-plugin-torch` in another project. Any chance it could be extracted into it's own repository and published to pypi?

Thanks!

**Comment by tgolsson** [comment]
That doesn't seem like an unreasonable request. I sort-of wanted https://github.com/pdm-project/pdm/issues/1461 to be resolved first, but at @frostming has reasonable concerns about how it'd work.

**Comment by frostming** [comment]
@tgolsson since you are already using `post_install` hook, I don't see anything stopping you from uploading it to pypi and changing the script to `pdm self add pdm-plugin-torch`.

@dsully If you want to use it immediately try
```
pdm self add "pdm-plugin-torch @ git+https://github.com/EmbarkStudios/emote.git#subdirectory=plugins/pdm-plugin-torch"
```

**Comment by tgolsson** [comment]
Oh for sure - I'm just saying it'd be nicer if there was first-class support for this kind of plugin with a dedicated section/sanctioned config formats. :-) I'm looking at publishing this in a separate repo and to PyPi right now and I'll mention that trick. Just feels a bit hacky.

**Comment by dsully** [comment]
@frostming - This plugin is what I was trying to make work in https://github.com/pdm-project/pdm/issues/1703, which fails on https://github.com/EmbarkStudios/emote/blob/main/plugins/pdm-plugin-torch/src/pdm_plugin_torch/main.py#L358 (also the same loop in the `handle_install` method.

**Comment by tgolsson** [comment]
Yeah that's a use-case I've never considered. My experience coming from a Tensorflow-heavy workflow is that even patch versions will have random regressions. So - we hard-pin exact versions. In fairness we haven't seen those issues with torch yet. 

I do think there is one ambiguous point in PEP-440 which is whether `torch==1.*+cu117` is valid. It does say the following:

> It is invalid to have a prefix match containing a development or local release such as 1.0.dev1.* or 1.0+foo1.*. If present, the development release segment is always the final segment in the public version, and the local version is ignored for comparison purposes, so using either in a prefix match wouldn’t make any sense.

And from the *example* given I'm lead to believe that the local version cannot be *in the prefix*, but it might just be a bad example.

FWIW I've got a repository set up with the plugin alone, just waiting on final checks. We should likely move any issues with this plugin there. I have an idea of doing a resolve without local versions, and then pinning everything else on that. However; that assumes that all versions exist for all APIs, which I'm not sure is true.

**Comment by dsully** [comment]
@frostming Any thoughts on how to proceed here? Thanks

**Comment by dsully** [comment]
Ping
